doc/lua_api.md

```diff
@@ -11368,6 +11368,16 @@ Functions: bit.tobit, bit.tohex, bit.bnot, bit.band, bit.bor, bit.bxor, bit.lshi
 
 See http://bitop.luajit.org/ for advanced information.
 
+Tracy Profiler
+--------------
+
+Minetest can be built with support for the Tracy profiler, which can also be
+useful for profiling mods and is exposed to Lua as the global `tracy`.
```

Comment:
IMO it would also be nice to provide `tracy.*` stubs (looking at the Tracy documentation, these can probably be implemented as no-op) even for builds without Tracy. That would allow mods to use tracy without having to put `if tracy ... end` all over the place.

Reply:
That probably introduces more code bloat than what's worth it.

Reply:
I don't think that would be very heavy in terms of code bloat, and it's kind of necessary to make Tracy usage in Lua comfortable.

Assuming the Tracy API is essentially a table of functions, something like the following should work:

```lua
local function no_op() end
tracy = tracy or setmetatable({}, {__index = function() return no_op end})
```

(The only problem with this is that it is a bit too permissive. Explicitly implementing no-ops for all Tracy API functions is probably a better idea.)

Reply:
I had also thought about this. But I specifically didn't add these stubs, because otherwise mods could rely on them.
If we don't add them, mods always have to check for the existence of `tracy`, or add their own stubs, or don't release their mods with tracy calls.
If we do add them, mods can release with them unconditionally, and if they do, they're not minetest api compliant. And if they do want to adhere to the api docs, they have to do the same checks as in the case where we didn't add the stubs. => I see no gain.

Reply:
We could officially add the existence of the stubs to the api though.
But idk if we want such an api change. And we could always add it later.